internal/server/subscriptionlistener.go

```diff
@@ -27,6 +28,11 @@ func (s *Server) StartSubscriptionListener() error {
 		return err
 	}
 
+	if device.AdminState == models.Locked || device.OperatingState == models.Down {
+		s.sdk.LoggingClient().Warnf("subscription listener not started for [%s]: device is locked or down", s.deviceName)
+		return nil
+	}
```

Comment:
Say we have a subscription collecting data from an OPC UA server. If the device becomes Locked and/or Down while this subscription exists, does it ever stop? It looks like it only checks once when the subscription is created.

Reply:
Good question - the answer is yes. The device driver's UpdateDevice callback function is triggered, which calls to this function to clean up old subscriptions and restart the subscription listener.

Reply:
I see it now. Thanks.